src/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -14,8 +14,28 @@ TODO: subsume existing test, rewrite fixture as needed.
 `;
 
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { assert } from '../../../../../common/util/util.js';
+import { kSamplerBindingTypes, kBufferBindingTypes } from '../../../../capability_info.js';
 import { ValidationTest } from '../../validation_test.js';
 
+function getBindGroupLayouts(
+  device: GPUDevice,
+  bindGroups: Array<Array<GPUBindGroupLayoutEntry>>
+): Array<GPUBindGroupLayout> {
+  const bindGroupLayouts = [];
+  for (let i = 0; i < bindGroups.length; ++i) {
+    // Support empty bindGroupLayout
+    const entries = [];
+    for (let j = 0; j < bindGroups[i].length; ++j) {
+      const binding = bindGroups[i][j];
+      assert(binding !== undefined);
+      entries.push(bindGroups[i][j]);
+    }
+    bindGroupLayouts.push(device.createBindGroupLayout({ entries }));
```

Comment:
why can't `bindGroups[i]` be passed directly here?

could this whole helper just be `bindGroups.map(entries => device.createBindGroupLayout({ entries }))`?
(and then maybe inline it instead of having a helper function)

Reply:
Cool!